ambry-network/src/main/java/com.github.ambry.network/Selector.java

```diff
@@ -556,16 +590,20 @@ private void read(SelectionKey key, Transmission transmission) throws IOExceptio
         transmission.onReceiveComplete();
```

Comment:
let's have this method return the NetworkReceive if `readComplete` and then use the future result to modify completedReceives from the main thread.